include/util/dplugin.h

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2017 - 2023 UnionTech Software Technology Co., Ltd.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef DPLUGIN_H
+#define DPLUGIN_H
+
+#include "dtkcore_global.h"
+
+#include <QLibrary>
+
+#include <memory>
+
+DCORE_BEGIN_NAMESPACE
+
+class DPluginPrivate;
+
+class DPlugin : public QObject
```

Comment:
Should DPlugin derive from DObject? If yes, it's better to use D_DECLARE_PRIVATE. And remember to make DPluginPrivate derive from DObjectPrivate.